alvr/server_openvr/cpp/alvr_server/Controller.h

```diff
@@ -54,6 +54,7 @@ class Controller : public TrackedDevice, public vr::ITrackedDeviceServerDriver {
     vr::EVRSkeletalTrackingLevel m_skeletonLevel;
 
     vr::DriverPose_t m_pose;
+    vr::VRBoneTransform_t m_tempHandBones[SKELETON_BONE_COUNT];
```

Comment:
This is just kinda not a good idea?? Not reusing variables is much better for code complexity, so the only reason this would be done is to improve performance.
Never mind that this is a horrible case of premature optimization, it's also just kinda useless. Stack allocations are insanely fast, like a single addition fast in the norm case. Which also doesn't matter because the compiler predetermines the stack layout and likely just allocates all of it at once.

In fact, this might actually be worse off for performance as the stack memory is usually permanently in cache, so putting it into a not as frequently accessed part of memory might worsen memory access times.  Also since the compiler has to deal with that being persistent memory, it'll likely do a worse job at optimizing.

Now none of those are absolute claims as I neither benchmarked this nor looked at the assembly, but as it stands it's just bad practice and you should undo it.

Reply:
hm good points